draft-thoji-scone-trone-protocol.md

```diff
@@ -305,17 +336,72 @@ packet and replace an existing rate signal.
 
 ~~~ pseudocode
 is_long = packet[0] & 0x80 == 0x80
-is_trone = compare(packet[1..5], TRONE_VERSION)
-if is_long and is_trone:
-  dcid_len = packet[5]
-  offset = 6 + dcid_len
-  scid_len = packet[offset]
-  offset = offset + 1 + scid_len
-  packet_rate = read_rate_signal(packet[offset : offset + rate_signal_size])
-  if target_rate.is_lower_than(packet_rate):
-    write_rate_signal(packet[offset : offset + rate_signal_size], target_rate)
+packet_version = packet[1..5]
+if is_long and (packet_version == TRONE1_VERSION or packet_version == TRONE2_VERSION):
+  target_rate_value = convert_rate_to_signal(target_rate, packet_version)
+  packet_rate_value = packet[0] & 0x3f
+  if packet_rate_value == 0x3f or target_rate_value < packet_rate_value:
+    packet[0] = packet[0] & 0xc0 | target_rate_value
 ~~~
 
+For TRONE packets with two versions, a network element applies the same rate
+policy to both, accounting for the different scales of each version. When a
```

Comment:
This reads a bit like a single TRONE packet can have two versions, perhaps rewrite as:
``````suggestion
A network element applies the same rate policy to TRONE packets of both versions, accounting for the different scales of each version. When a
```